src/web/jcc/client/components/__tests__/Accordion.test.tsx

```diff
@@ -0,0 +1,58 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import { Accordion, AccordionSummary, AccordionDetails, Typography } from "@mui/material";
+
+// Mock the onChange function, so we can track calls to it
+const mockOnChange = jest.fn();
+
+interface Props {
+    onChange: () => void;
+}
+
+const mockProps: Props = { onChange: mockOnChange };
+
+function AccordionTestComponent(props: Props) {
+    return (
+        <Accordion expanded={false} onChange={props.onChange}>
+            <AccordionSummary>
+                <Typography>Accordion Title Here</Typography>
+            </AccordionSummary>
+            <AccordionDetails>Accordion Details Here</AccordionDetails>
+        </Accordion>
+    );
+}
+
+describe("Accordion Tests", () => {
+    test("Accordion Test", async () => {
+        // Render the component
+        render(<AccordionTestComponent {...mockProps} />);
+
+        // Get the accordion elements
+        const summary = screen.getByText("Accordion Title Here");
+        const details = screen.getByText("Accordion Details Here");
+
+        //Summary should be visible
+        expect(summary).toBeVisible();
+        // Accordion starts closed, so details invisible
+        expect(details).not.toBeVisible();
+
+        // Click the summary
+        await userEvent.click(summary);
+
+        // Wait for onChange logic to be triggered, which opens the details
+        waitFor(() => {
```

Comment:
Should we add await before this? The waitFor calls are not awaited, which could lead to unexpected tests if the state changes asynchronously

Reply:
You're right!  But, in this case it turns out I don't even need waitFor, (the await userEvent.click does the required waiting).

There was another problem with the code that caused me to resort, incorrectly, to `waitFor`.

I need to use the `isExpanded` state variable to control the opening and closing of the Accordion, (before I had assumed it was internal state to the widget, but it is not).

By using `waitFor` without `await`, I made the test "pass" by not actually completing the `waitFor` callback!  So yeah, we must always `await` the `waitFor` calls.